remote/remote.go

```diff
@@ -207,11 +207,24 @@ func newV1Image(keychain authn.Keychain, repoName string, platform imgutil.Platf
 			}
 			return nil, fmt.Errorf("connect to repo store '%s': %s", repoName, err.Error())
 		}
+		break
 	}
-
 	return image, nil
 }
 
+func isRetryable(err error) bool {
+	if err == io.EOF {
+		return true
+	}
+	if transportErr, ok := err.(*transport.Error); ok {
+		switch transportErr.StatusCode {
+		case http.StatusNotFound, http.StatusUnauthorized:
+			return true
+		}
```

Comment:
```suggestion
		if transportErr.StatusCode ==  http.StatusNotFound || transportErr.StatusCode == http.StatusUnauthorized {
			return true
		}
```

I think that we usually use `switch-case` when for each result, we should act differently.
You can also have only one "big" if condition for all of this `transportErr`, but I'm not sure whether it'll be more readable. I don't have strong opinion on whether it should be one big if or two nested ifs.

Reply:
I see that in the `if` condition on line 199, we also check that `len(transportErr.Errors) > 0`.
Do we really need it? And if we do, do we need to add it to the condition on line 219?

Reply:
Honestly looking both options if statement looks for readable to me

Reply:
Sorry if this was already covered, but this has been confirmed that we're actually seeing `401` and `404` and then _retrying_ the requests makes these go away?

I'm asking because `40x` errors are usually caused by the client, not by the server or by network interference.

Reply:
Well, the only way I managed to reproduce the error mentioned by @jabrown85  in the [ticket 1069](https://github.com/buildpacks/pack/issues/1069) was having the newV1Image method returning an empty image, and that happens when we get 404, 401 or an error with the message "no child with platform", I am not very happy to retry with 40x but otherwise not sure how to address the issue

Reply:
I see. Well, works for me 🙂

Reply:
@yaelharel I just added the suggestion about the if statement

Reply:
@micahyoung do you have any thoughts here about the 40x errors?

Reply:
> the only way I managed to reproduce the error ... was having the newV1Image method returning an empty image

If you're able to wait until https://github.com/buildpacks/imgutil/pull/110 is merged (or willing to retrofit), I found it's pretty easy to wrap a `ggcr/registry` with custom request/response logic like I do in my PR to get read-only behavior:
https://github.com/buildpacks/imgutil/blob/f27928a6bb8e4ddf6694f55baa3d7413efba6c6b/testhelpers/docker_registry.go#L77-L87

So you could potentially make a new "flaky" handler like:
```go
func DropFirstRequest(handler http.Handler) http.Handler {
        firstRequest := true
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if firstRequest {
                         firstRequest = false
			return
		}

		handler.ServeHTTP(w, r)
	})
}
``` 

From which you could make a new "flaky" read-only registry: 
```go
dropFirstRequestHandler := DropFirstRequest(sharedRegistryHandler)
flaklyDockerRegistry = h.NewDockerRegistry(h.WithSharedHandler(dropFirstRequestHandler))
flaklyDockerRegistry.Start(t)
defer flaklyDockerRegistry.Stop(t)
```